cms/djangoapps/contentstore/views/videos.py

```diff
@@ -424,28 +546,39 @@ def videos_index_html(course):
-            'image_upload_url': reverse_course_url('video_images_handler', unicode(course.id)),
-            'video_handler_url': reverse_course_url('videos_handler', unicode(course.id)),
-            'encodings_download_url': reverse_course_url('video_encodings_download', unicode(course.id)),
-            'default_video_image_url': _get_default_video_image_url(),
-            'previous_uploads': _get_index_videos(course),
-            'concurrent_upload_limit': settings.VIDEO_UPLOAD_PIPELINE.get('CONCURRENT_UPLOAD_LIMIT', 0),
-            'video_supported_file_formats': VIDEO_SUPPORTED_FILE_FORMATS.keys(),
-            'video_upload_max_file_size': VIDEO_UPLOAD_MAX_FILE_SIZE_GB,
-            'video_image_settings': {
-                'video_image_upload_enabled': WAFFLE_SWITCHES.is_enabled(VIDEO_IMAGE_UPLOAD_ENABLED),
-                'max_size': settings.VIDEO_IMAGE_SETTINGS['VIDEO_IMAGE_MAX_BYTES'],
-                'min_size': settings.VIDEO_IMAGE_SETTINGS['VIDEO_IMAGE_MIN_BYTES'],
-                'max_width': settings.VIDEO_IMAGE_MAX_WIDTH,
-                'max_height': settings.VIDEO_IMAGE_MAX_HEIGHT,
-                'supported_file_formats': settings.VIDEO_IMAGE_SUPPORTED_FILE_FORMATS
-            }
+    context = {
+        'context_course': course,
+        'image_upload_url': reverse_course_url('video_images_handler', unicode(course.id)),
+        'video_handler_url': reverse_course_url('videos_handler', unicode(course.id)),
+        'encodings_download_url': reverse_course_url('video_encodings_download', unicode(course.id)),
+        'default_video_image_url': _get_default_video_image_url(),
+        'previous_uploads': _get_index_videos(course),
+        'concurrent_upload_limit': settings.VIDEO_UPLOAD_PIPELINE.get('CONCURRENT_UPLOAD_LIMIT', 0),
+        'video_supported_file_formats': VIDEO_SUPPORTED_FILE_FORMATS.keys(),
+        'video_upload_max_file_size': VIDEO_UPLOAD_MAX_FILE_SIZE_GB,
+        'video_image_settings': {
+            'video_image_upload_enabled': WAFFLE_SWITCHES.is_enabled(VIDEO_IMAGE_UPLOAD_ENABLED),
+            'max_size': settings.VIDEO_IMAGE_SETTINGS['VIDEO_IMAGE_MAX_BYTES'],
+            'min_size': settings.VIDEO_IMAGE_SETTINGS['VIDEO_IMAGE_MIN_BYTES'],
+            'max_width': settings.VIDEO_IMAGE_MAX_WIDTH,
+            'max_height': settings.VIDEO_IMAGE_MAX_HEIGHT,
+            'supported_file_formats': settings.VIDEO_IMAGE_SUPPORTED_FILE_FORMATS
         }
-    )
+    }
+
+    if WAFFLE_SWITCHES.is_enabled(THIRD_PARTY_TRANSCRIPTION_ENABLED):
+        context.update({
+            'third_party_transcript_settings': {
+                'transcript_preferences_handler_url': reverse_course_url(
+                    'transcript_preferences_handler',
+                    unicode(course.id)
+                ),
+                'transcription_plans': get_3rd_party_transcription_plans(),
+            },
+            'active_transcript_preferences': get_transcript_preferences(unicode(course.id))
+        })
```

Comment:
Well, i think instead of making lists of different thing in js, let's send those lists from backend, like providers, fidelity, turnaround languages.